lib/galaxy/webapps/galaxy/api/groups.py

```diff
@@ -2,61 +2,85 @@
 from galaxy.managers.groups import GroupsManager
 from galaxy.schema.fields import EncodedDatabaseIdField
-from galaxy.web import (
-    expose_api,
-    require_admin,
+from galaxy.schema.groups import (
+    GroupCreatePayload,
+    GroupIndexListResponse,
+    GroupShowResponse,
 )
 from galaxy.webapps.galaxy.api import (
-    BaseGalaxyAPIController,
     depends,
+    DependsOnTrans,
+    Router,
 )
 
 log = logging.getLogger(__name__)
 
+router = Router(tags=["groups"])
+
 
-class GroupAPIController(BaseGalaxyAPIController):
-    manager = depends(GroupsManager)
+@router.cbv
+class FastAPIGroups:
+    manager: GroupsManager = depends(GroupsManager)
 
-    @expose_api
-    @require_admin
-    def index(self, trans: ProvidesAppContext, **kwd):
-        """
-        GET /api/groups
-        Displays a collection (list) of groups.
-        """
+    @router.get(
+        "/api/groups",
+        summary="Displays a collection (list) of groups.",
+        require_admin=True,
+    )
+    def index(
+        self,
+        trans: ProvidesAppContext = DependsOnTrans,
+    ) -> GroupIndexListResponse:
+        """GET /api/groups - Displays a collection (list) of groups."""
         return self.manager.index(trans)
 
-    @expose_api
-    @require_admin
-    def create(self, trans: ProvidesAppContext, payload: Dict[str, Any], **kwd):
-        """
-        POST /api/groups
-        Creates a new group.
-        """
+    @router.post(
+        "/api/groups",
+        summary="Creates a new group.",
+        require_admin=True,
+    )
+    def create(
+        self,
+        trans: ProvidesAppContext = DependsOnTrans,
+        payload: GroupCreatePayload = Body(),
+    ) -> GroupIndexListResponse:
+        """POST /api/groups - Creates a new group."""
         return self.manager.create(trans, payload)
 
-    @expose_api
-    @require_admin
-    def show(self, trans: ProvidesAppContext, id: EncodedDatabaseIdField, **kwd):
-        """
-        GET /api/groups/{encoded_group_id}
-        Displays information about a group.
-        """
-        return self.manager.show(trans, EncodedDatabaseIdField.decode(id))
-
-    @expose_api
-    @require_admin
-    def update(self, trans: ProvidesAppContext, id: EncodedDatabaseIdField, payload: Dict[str, Any], **kwd):
-        """
-        PUT /api/groups/{encoded_group_id}
-        Modifies a group.
-        """
-        self.manager.update(trans, EncodedDatabaseIdField.decode(id), payload)
+    @router.get(
+        "/api/groups/{encoded_group_id}",
+        summary="Displays information about a group.",
+        require_admin=True,
+    )
+    def show(
+        self,
+        trans: ProvidesAppContext = DependsOnTrans,
+        encoded_group_id: EncodedDatabaseIdField = Path(),
+    ) -> GroupShowResponse:
+        """GET /api/groups/{encoded_group_id} - Displays information about a group."""
+        group_id = EncodedDatabaseIdField.decode(encoded_group_id)
```

Comment:
Use a `DecodedDatabaseIdField` for any input parameter or payload model field, this will validate and auto-decode the ID so you don't need to manually decode it like here.